routes/profiles.rb

```diff
@@ -50,13 +51,29 @@ def self.registered(app)
           end
 
           update_profile = lambda do
-            if !@profile || @profile[:role] != 'superadmin'
-              halt 400, 'Parameter requirements were not met.' unless check_parameters(params, profile_params)
+            updatedProfile = params[:updatedProfile]
+            if !@profile || @profile[:id].to_s != updatedProfile[:_id][:$oid].to_s
+              if @profile[:role] != 'superadmin'
+                halt 400, 'Permission Denied.' unless check_parameters(params, profile_params)
+              end
+            end
+            if !check_parameters(updatedProfile, profile_params)
```

Comment:
We can compact this down to one line - `halt 400, 'Parameter requirements were not met.' unless check_parameters(updatedProfile, profile_params)`. Then we can get rid of the `else`.

Reply:
Good to know. I have not used this before.